cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -214,14 +209,23 @@ exports.handler = async function (event, context) {
                     aliases: aliases,
                     idempotencyToken: CRYPTO
                         .createHash("md5")
-                        .update(physicalResourceID)
+                        .update(`/${serviceName}/${aliasesSorted}`)
                         .digest("hex")});
+                physicalResourceID = certificateARN; // Update the physical resource ID if a new certificate is created.
                 const options = await waitForValidationOptionsToBeReady(certificateARN, aliases);
                 await activate(options, certificateARN, loadBalancerDNS, loadBalancerHostedZoneID);
                 break;
             case "Delete":
-                let unusedOptions = await unusedValidationOptions(aliases, loadBalancerDNS);
-                await deleteCertificate(aliases);
+                if (!physicalResourceID.startsWith("arn:")) {
+                    // This means no certificate has been created, nor any records. Exit without doing anything.
```

Comment:
would this cause a weird exception if `physicalResourceID` is `undefined`?

Reply:
From what I have observed, the `physicalResourceID` would be provided as a weird looking random string (e.g. "a3dkdfanrdsa3") to the "Delete" event when the creation fails. However I can't find the documentation for this behavior, so I added an extra logic that says: `if (!physicalResourceID || !physicalResourceID.startsWith("arn:"))` to deal with the possibility when the physical resource ID is undefined.